java/code/src/com/suse/oval/parser/OvalParser.java

```diff
@@ -0,0 +1,724 @@
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.suse.oval.parser;
+
+import com.redhat.rhn.domain.rhnpackage.PackageType;
+
+import com.suse.oval.exceptions.OvalParserException;
+import com.suse.oval.ovaltypes.Advisory;
+import com.suse.oval.ovaltypes.AdvisoryAffectedType;
+import com.suse.oval.ovaltypes.AdvisoryCveType;
+import com.suse.oval.ovaltypes.AdvisoryResolutionType;
+import com.suse.oval.ovaltypes.ArchType;
+import com.suse.oval.ovaltypes.BaseCriteria;
+import com.suse.oval.ovaltypes.CriteriaType;
+import com.suse.oval.ovaltypes.CriterionType;
+import com.suse.oval.ovaltypes.DefinitionClassEnum;
+import com.suse.oval.ovaltypes.DefinitionType;
+import com.suse.oval.ovaltypes.EVRType;
+import com.suse.oval.ovaltypes.LogicOperatorType;
+import com.suse.oval.ovaltypes.MetadataType;
+import com.suse.oval.ovaltypes.ObjectType;
+import com.suse.oval.ovaltypes.OperationEnumeration;
+import com.suse.oval.ovaltypes.OvalRootType;
+import com.suse.oval.ovaltypes.StateType;
+import com.suse.oval.ovaltypes.TestType;
+import com.suse.oval.ovaltypes.VersionType;
+import com.suse.oval.ovaltypes.linux.DpkginfoObject;
+import com.suse.oval.ovaltypes.linux.DpkginfoState;
+import com.suse.oval.ovaltypes.linux.DpkginfoTest;
+import com.suse.oval.ovaltypes.linux.RpminfoObject;
+import com.suse.oval.ovaltypes.linux.RpminfoState;
+import com.suse.oval.ovaltypes.linux.RpminfoTest;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.codehaus.stax2.XMLEventReader2;
+import org.codehaus.stax2.evt.XMLEvent2;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.StartElement;
+
+/**
+ * The Oval Parser is responsible for parsing OVAL(Open Vulnerability and Assessment Language) documents
+ */
+public class OvalParser {
+    public static final int DEFINITIONS_BULK_SIZE = 500;
+
+    private static final Logger LOG = LogManager.getLogger(OvalParser.class);
+    private static final List<String> TEST_TYPES = List.of("rpminfo_test", "dpkginfo_test");
+    private static final List<String> OBJECT_TYPES = List.of("rpminfo_object", "dpkginfo_object");
+    private static final List<String> STATE_TYPES = List.of("rpminfo_state", "dpkginfo_state");
+
+    /**
+     * Parse the given OVAL file
+     *
+     * @param ovalFileURL the OVAL file to parse
+     * @return the parsed OVAL encapsulated in an {@link OvalRootType} object.
+     * */
+    public OvalRootType parse(URL ovalFileURL) throws OvalParserException {
```

Comment:
This parser is only used for testing. Should we move it to a test-only class? The method could be a utility that would receive a `OvalParser` instance and the File location. Since it keeps all in memory, I think it is risky to have in the main class when is used only for testing.